sql/src/test/java/io/crate/analyze/InsertFromSubQueryAnalyzerTest.java

```diff
@@ -182,65 +183,98 @@ public void testImplicitTypeCasting() throws Exception {
+            }
         }
     }
 
     @Test
     public void testFromQueryWithOnDuplicateKeyValues() throws Exception {
-        InsertFromSubQueryAnalyzedStatement statement =
-            e.analyze("insert into users (id, name) (select id, name from users) " +
-                      "on duplicate key update name = substr(values (name), 1, 1)");
-
-        Assert.assertThat(statement.onDuplicateKeyAssignments().size(), is(1));
-
-        for (Map.Entry<Reference, Symbol> entry : statement.onDuplicateKeyAssignments().entrySet()) {
-            assertThat(entry.getKey(), isReference("name"));
-            assertThat(entry.getValue(), isFunction(SubstrFunction.NAME));
-            Function function = (Function) entry.getValue();
-            assertThat(function.arguments().get(0), instanceOf(InputColumn.class));
-            InputColumn inputColumn = (InputColumn) function.arguments().get(0);
-            assertThat(inputColumn.index(), is(1));
-            assertThat(inputColumn.valueType(), instanceOf(StringType.class));
+        String[] insertStatements = new String[2];
+        insertStatements[0] = "insert into users (id, name) (select id, name from users) " +
+                              "on duplicate key update name = substr(values (name), 1, 1)";
+        insertStatements[1] = "insert into users (id, name) (select id, name from users) " +
+                              "on conflict do update set name = substr(excluded.name, 1, 1)";
+
+        for (String insertStatement : insertStatements) {
+            InsertFromSubQueryAnalyzedStatement statement = e.analyze(insertStatement);
+
+            Assert.assertThat(statement.onDuplicateKeyAssignments().size(), is(1));
+
+            for (Map.Entry<Reference, Symbol> entry : statement.onDuplicateKeyAssignments().entrySet()) {
+                assertThat(entry.getKey(), isReference("name"));
+                assertThat(entry.getValue(), isFunction(SubstrFunction.NAME));
+                Function function = (Function) entry.getValue();
+                assertThat(function.arguments().get(0), instanceOf(InputColumn.class));
+                InputColumn inputColumn = (InputColumn) function.arguments().get(0);
+                assertThat(inputColumn.index(), is(1));
+                assertThat(inputColumn.valueType(), instanceOf(StringType.class));
+            }
         }
     }
 
     @Test
     public void testFromQueryWithUnknownOnDuplicateKeyValues() throws Exception {
-        expectedException.expect(ColumnUnknownException.class);
-        expectedException.expectMessage("Column does_not_exist unknown");
-        e.analyze("insert into users (id, name) (select id, name from users) " +
-                  "on duplicate key update name = values (does_not_exist)");
+        try {
+            e.analyze("insert into users (id, name) (select id, name from users) " +
+                      "on duplicate key update name = values (does_not_exist)");
```

Comment:
I think we should add some `fail` calls here so that it doesn't pass in case it isn't throwing an exception.

(also in the other places)

Reply:
Of course, totally missed that when I converted from the expectedException syntax.